src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,485 @@
 
 -- | Extract the underlying relation.
--- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
-toRelation = symmetricClosure . fromSymmetric
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- toRelation ('edge' 1 2)                           == 'Algebra.Graph.Relation.edges' [(1,2), (2,1)]
+-- 'Algebra.Graph.Relation.vertexCount' . toRelation == 'vertexCount'
+-- 'Algebra.Graph.Relation.edgeCount' . toRelation   <= (* 2) . 'edgeCount'
+-- @
+toRelation :: SymmetricRelation a -> R.Relation a
+toRelation = fromSymmetric
+
+-- | Construct the graph comprising /a single edge/.
+-- Complexity: /O(1)/ time, memory and size.
+--
+-- @
+-- edge x y                 == edges [(x,y), (y,x)]
+-- edge x y                 == connect (vertex x) (vertex y)
+-- edge x y                 == edge y x
+-- 'hasEdge' x y (edge x y) == True
+-- 'edgeCount'   (edge x y) == 1
+-- 'vertexCount' (edge 1 1) == 1
+-- 'vertexCount' (edge 1 2) == 2
+-- @
+edge :: Ord a => a -> a -> SymmetricRelation a
+edge x y = SymmetricRelation $ RI.Relation (Set.fromList [x, y]) (Set.fromList [(x, y), (y,x)]) 
+
+-- | Construct the graph comprising a given list of isolated vertices.
+-- Complexity: /O(L * log(L))/ time and /O(L)/ memory, where /L/ is the length
+-- of the given list.
+--
+-- @
+-- vertices []              == 'empty'
+-- vertices [x]             == 'vertex' x
+-- 'hasVertex' x . vertices == 'elem' x
+-- 'vertexCount' . vertices == 'length' . 'Data.List.nub'
+-- 'vertexSet'   . vertices == Set.'Set.fromList'
+-- @
+vertices :: Ord a => [a] -> SymmetricRelation a
+vertices = SymmetricRelation . R.vertices
+
+-- | Construct the graph from a list of edges.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- edges []            == 'empty'
+-- edges [(x,y)]       == 'edge x y'
+-- 'edgeCount' . edges == 'length' . 'Data.List.nub' . 'uncurry' 'Data.List.union' . '<map swap, id>'
+-- @
+edges :: Ord a => [(a, a)] -> SymmetricRelation a
+edges es = SymmetricRelation $ RI.Relation (Set.fromList $ uncurry (++) $ unzip es) (Set.fromList es `Set.union` Set.map swap (Set.fromList es))
+
+-- | Overlay a given list of graphs.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- overlays []          == 'empty'
+-- overlays [x]         == x
+-- overlays [x,y]       == 'overlay' x y
+-- overlays             == 'foldr' 'overlay' 'empty'
+-- 'isEmpty' . overlays == 'all' 'isEmpty'
+-- @
+overlays :: Ord a => [SymmetricRelation a] -> SymmetricRelation a
+overlays = SymmetricRelation . R.overlays . map fromSymmetric
+
+-- | Connect a given list of graphs.
+-- Complexity: /O((n + m) * log(n))/ time and /O(n + m)/ memory.
+--
+-- @
+-- connects []          == 'empty'
+-- connects [x]         == x
+-- connects [x,y]       == 'connect' x y
+-- connects             == 'foldr' 'connect' 'empty'
+-- 'isEmpty' . connects == 'all' 'isEmpty'
+-- connects             == connects . 'reverse
```

Comment:
`reverse` is missing the closing quote. Please make sure Haddock gives you no warnings for this file.

Reply:
Thanks for noticing, I always forget to check haddock. I'm sorry.